DPE-7166: Add spread tests and backward compatibility with legacy OCI

taurus-forever opened the pull request:
* add backward compatibility with legacy OCI and official DockerHub image
* add spread-based test using `rockctraft test`
* add `rockctraft test` to CI
* use Lint CI from data-platform-workflows
* update readme with more examples

taurus-forever committed:
DPE-7166: Add spread tests and backward compatibility with legacy OCI

* add backward compatibility with legacy OCI and official DockerHub image
* add spread-based test using `rockctraft test`
* add `rockctraft test` to CI
* use Lint CI from data-platform-workflows
* update readme with more examples

taurus-forever committed:
Add arm tests + bump data-platform-workflows

taurus-forever committed:
Update README.md

taurus-forever commented on rockcraft.yaml:
```diff
@@ -11,8 +11,9 @@ platforms:
 environment:
-    TZ: UTC
     PATH: "/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/lib/postgresql/16/bin"
+    PGDATA: /var/lib/postgresql/data
```
[Backward compatibility #1](https://git.launchpad.net/~ubuntu-docker-images/ubuntu-docker-images/+git/postgresql/tree/Dockerfile?h=14-22.04#n81)

taurus-forever commented on rockcraft.yaml:
```diff
@@ -60,6 +61,8 @@ parts:
             chown -R 584788 $CRAFT_PRIME/var/run/postgresql
             # Docker entrypoint compatibility
             mkdir -p $CRAFT_PRIME/docker-entrypoint-initdb.d/
+            # Allow access to PostgreSQL
+            sed -ri "s!^#?(listen_addresses)\s*=\s*\S+.*!\1 = '*'!" $CRAFT_PRIME/usr/share/postgresql/16/postgresql.conf.sample
```
[Backward compatibility #2](https://git.launchpad.net/~ubuntu-docker-images/ubuntu-docker-images/+git/postgresql/tree/Dockerfile?h=14-22.04#n72)

taurus-forever committed:
Fix shellcheck actionlint warnings

taurus-forever committed:
Fix shellcheck actionlint warnings #2

taurus-forever committed:
Fix shellcheck actionlint warnings #3

taurus-forever committed:
Increase timeout on ARM (slow runners)

sinclert-canonical commented on .github/workflows/ci.yaml:
```diff
@@ -12,16 +12,47 @@ on:
     name: Lint
-    runs-on: ubuntu-latest
-    timeout-minutes: 5
+    uses: canonical/data-platform-workflows/.github/workflows/lint.yaml@v32.2.2
```
Good stuff. Will port changes into MySQL rocks 👍🏻

sinclert-canonical commented on spread/.extension:
If I understood you correctly, this script has been copied from the initialization step of... LXD? SnapD?

It would be good to know where it comes from, in order to be able to see how the source of truth evolves, and copy improvements / extensions. A simple comment at the top of the file would be enough.

taurus-forever replied:
It is being created via `rockcraft init --profile=test`. 
This is the current approach for all `*craft test` tools to make spread.yaml file simpler (compare it to [mysql-operator](https://github.com/canonical/mysql-operator/blob/main/spread.yaml#L8)).

taurus-forever committed:
DPE-7166: Removed --raw-output from oci-factory release

It is no longer necessary as we have removed quotes from version field
of rockcraft.yaml, discussed this with Rock team, it should be safe.

taurus-forever committed:
DPE-7166: Cast yq type after removing quotes from rockcraft.yaml

Otherwise:

> yq '.version | split(".")[0]' rockcraft.yaml
Error: cannot split !!float, can only split strings

dragomirp commented on spread/tests/smoke/task.yaml:
We should most probably assert that the payload and rockraft.yaml verions match, since that's easy to forget, but it can be added in follow ups.

taurus-forever replied:
Excellent idea! Building a followup PR...

taurus-forever replied:
Created https://github.com/canonical/postgresql-rock/pull/3
